File: backend/spring-boot-book-ecommerce/src/main/java/com/luvannie/springbootbookecommerce/dto/Purchase.java
package com.luvannie.springbootbookecommerce.dto;

import com.luvannie.springbootbookecommerce.entity.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;
@Data
public class Purchase {

    private Customer customer;

    private Address shippingAddress;

    private Order order;

    private Set<OrderItem> orderItems;

    private User user;


}

File: frontend/angular-book-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer | undefined;
    shippingAddress: Address | undefined;
    order: Order | undefined;
    orderItems: OrderItem[] | undefined;
    user_id: number | undefined;
    shipping_method: string| undefined;
    payment_method: string| undefined;
    // coupon_code: string| undefined;

    
}
